**project** **goal**:

apply classification and clustering to predict the presence of heart disease

Classification:  
  The dataset includes a target attribute called **"target"** with two classes:  
  - **0** = No Heart Disease  
  - **1** = Heart Disease

Clustering:  
  The clustering task will group patients with similar characteristics **( age, cholesterol level,
  maximum heart rate)**. This helps us discover hidden patterns in the data, such as groups of patients
  that share common health risk factors


**Dataset Source**:

the dataset is taken from kaggle
[Heart Disease Datase](https://www.kaggle.com/datasets/johnsmith88/heart-disease-dataset)

In [ ]:
from google.colab import files
uploaded = files.upload()


Saving heart.csv to heart.csv


In [ ]:
import pandas as pd

df = pd.read_csv("heart.csv")

print("Shape (rows, columns):", df.shape)

print("\nInfo about dataset:")
df.info()

print("\nClass distribution:")
print(df['target'].value_counts())

print("\nSample rows:")
print(df.head())


Shape (rows, columns): (1025, 14)

Info about dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1025 entries, 0 to 1024
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1025 non-null   int64  
 1   sex       1025 non-null   int64  
 2   cp        1025 non-null   int64  
 3   trestbps  1025 non-null   int64  
 4   chol      1025 non-null   int64  
 5   fbs       1025 non-null   int64  
 6   restecg   1025 non-null   int64  
 7   thalach   1025 non-null   int64  
 8   exang     1025 non-null   int64  
 9   oldpeak   1025 non-null   float64
 10  slope     1025 non-null   int64  
 11  ca        1025 non-null   int64  
 12  thal      1025 non-null   int64  
 13  target    1025 non-null   int64  
dtypes: float64(1), int64(13)
memory usage: 112.2 KB

Class distribution:
target
1    526
0    499
Name: count, dtype: int64

Sample rows:
   age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slo


## Dataset Description

- *Number of rows (instances):* 1,025  
- *Number of columns (attributes):* 14  

### Attributes (Features):
1. *age* → Age of the patient (**Numeric**)  
2. *sex* → Gender (1 = Male, 0 = Female) (**Binary**)  
3. *cp* → Chest pain type (0–3) (**Ordinal**)  
4. *trestbps* → Resting blood pressure (mm Hg) (**Numeric**)  
5. *chol* → Serum cholesterol (mg/dl) (**Numeric**)  
6. *fbs* → Fasting blood sugar (>120 mg/dl, 1 = True, 0 = False) (**Binary**)  
7. *restecg* → Resting electrocardiographic results (0–2) (**Nominal**)  
8. *thalach* → Maximum heart rate achieved (**Numeric**)  
9. *exang* → Exercise induced angina (1 = Yes, 0 = No) (**Binary**)  
10. *oldpeak* → ST depression induced by exercise (**Numeric**)  
11. *slope* → Slope of the peak exercise ST segment (0–2) (**Ordinal**)  
12. *ca* → Number of major vessels (0–4) (**Numeric – Discrete**)  
13. *thal* → Thalassemia (1 = normal, 2 = fixed defect, 3 = reversible defect) (**Nominal**)  
14. *target* → Heart disease presence (0 = No, 1 = Yes) (**Binary – Target**)  

### Target (Class Label):
- *0* = No heart disease (499 instances)  
- *1* = Heart disease (526 instances)  


Sample rows:
| age | sex | cp | trestbps | chol | fbs | restecg | thalach | exang | oldpeak | slope | ca | thal | target |
|-----|-----|----|----------|------|-----|---------|---------|-------|---------|-------|----|------|--------|
| 52  | 1   | 0  | 125      | 212  | 0   | 1       | 168     | 0     | 1.0     | 2     | 2  | 3    | 1      |
| 53  | 1   | 0  | 140      | 203  | 1   | 1       | 155     | 1     | 3.1     | 0     | 0  | 3    | 0      |
| 70  | 1   | 0  | 145      | 174  | 0   | 1       | 125     | 1     | 2.6     | 0     | 0  | 3    | 0      |
| 61  | 0   | 0  | 138      | 203  | 0   | 1       | 161     | 0     | 0.0     | 2     | 1  | 3    | 0      |
| 62  | 0   | 0  | 138      | 294  | 1   | 1       | 106     | 0     | 1.9     | 1     | 3  | 2    | 0      |
